static/js/util/validation.js

```diff
@@ -87,6 +87,26 @@ export function educationValidation(profile) {
   }
 }
 
+export function educationUiValidation(profile, ui) {
+  if (profile.education === undefined) {
+    profile = Object.assign({}, profile, {
+      education: []
+    });
+  }
+
+  let errors = {};
+  for (let {value, label} of EDUCATION_LEVELS) {
+    let items = profile.education.filter(education => education.degree_name === value);
+    if (ui.educationDegreeInclusions[value] && items.length === 0) {
+      errors[`education_${value}_required`] = `${label} is required if switch is set`;
+    }
+    if (!ui.educationDegreeInclusions[value] && items.length > 0) {
+      errors[`education_${value}_required`] = `${label} switch is off but there is still education history in card`;
+    }
+  }
+  return errors;
+}
```

Comment:
For the first one, I think it would be good to give the user some guidance on what to do, e.g.

"A Master's Degree is required if the Master's Degree switch is on. Click the + button to add a degree or switch it off"

... that's probably too long though.

What's the user supposed to do in the 2nd case, on L103?